File: Java/TP3_MET_JAVA/src/Iteratif/TailleIteratif.java
package Iteratif;

public class TailleIteratif implements VisiteurMotIteratif<Integer> {

    @Override
    public Integer casVide() {
        return 0;
    }

    @Override
    public Integer casCaractereMot(Character c, Integer m) {
        return m + 1;
    }
}

File: TypeScript/TP3_MET_TS/src/Iteratif/TailleIteratif.ts
import {VisiteurMotIteratif} from "./VisiteurMotIteratif";

export class TailleIteratif implements VisiteurMotIteratif<number> {

    public casVide(): number {
        return 0;
    }

    public casCaractereMot(c: string, m: number): number {
        return m + 1;
    }
}
